Automattic/jetpack
# Nav Unification: Add support for preferred view on individual screens

Fixes https://github.com/Automattic/wp-calypso/issues/54021
Fixes https://github.com/Automattic/wp-calypso/issues/53710

#### Changes proposed in this Pull Request:
Add support for storing whether the user prefers to see the default view (Calypso) or the classic view (WP Admin) of screens that have both views when accessed from the WP.com sidebar menu.

The preferred views are stored on a new `jetpack_admin_menu_preferred_views` user option, which currently can be updated only programmatically. On a follow-up PR (https://github.com/Automattic/wp-calypso/issues/53709) we'll make sure that the preferred views are updated based on user actions.

This PR also removes all the duplicate / advance menus we were registering as they are no longer need due to the quick switch.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Set up the testing environment:
  - Simple sites: Apply D63194-code to your WP.com sandbox, and sandbox both the API and a Simple site.
  - Atomic/Jetpack sites: Install Jetpack Beta and activate the `update/admin-menu-handle-preferred-screen` branch.
- Use the WP shell for updating the preferred views as you wish on a Simple or Atomic site (make sure to switch to your user and site):
```php
$preferred_views = get_user_option( 'jetpack_admin_menu_preferred_views' );
if ( ! $preferred_views ) {
	$preferred_views = array();
}
$preferred_views[ $screen ] = 'classic'; // or 'default'.
update_user_option( get_current_user_id(), 'jetpack_admin_menu_preferred_views', $preferred_views );
```
- where `$screen` can be any of the following:
  - `edit.php`: Posts > All posts
  - `edit-tags.php?taxonomy=category`: Posts > Categories
  - `edit-tags.php?taxonomy=post_tag`: Posts > Tags
  - `upload.php`: Media
  - `edit.php?post_type=page`: Pages > All pages
  - `edit.php?post_type=jetpack-testimonial`: Testimonials > All testimonials
  - `edit.php?post_type=jetpack-portfolio`:  Portfolio > All projects
  - `edit-comments.php`: Comments
  - `themes.php`: Appearance > Themes
  - `users.php`: Users > All users
  - `import.php`: Tools > Import
  - `export.php`: Tools > Export
  - `options-general.php`: Settings > General
  - `options-writing.php`: Settings > Writing
  - `options-discussion.php`: Settings > Discussion
- Go to wordpress.com.
- Switch to the site where the preferred views have been set on.
- Make sure that the menu/submenu items point to Calypso or WP Admin based on the preferred views.
- For menu/submenus items where a preferred view has not been set, make sure that they point to Calypso or WP Admin based on the global "show wp-admin pages" toggle.
- Make sure that the following exceptions are correctly handled:
  - Users > All users points to Calypso when no preferred view has been set, regardless of the global toggle
  - Settings > General points to Calypso when no preferred view has been set, regardless of the global toggle
  - (Simple sites only) Appearance > Themes points to Calypso when no preferred view has been set, regardless of the global toggle
  - (Atomic sites only) Tools > Export always points to WP Admin
  - (Atomic sites only) Appearance > Customize always points to WP Admin
- Make sure that the following duplicated / advanced menus are no longer registered:
  - Settings > Advanced General
  - Settings > Advanced Writing
  - Users > Advanced Users Management
  - Appearance > Add New Theme
- Switch to a Jetpack site.
- Make sure there are no changes in the menu when compared to production.
